src/input.cpp

```diff
@@ -882,18 +884,30 @@ void input_context::display_help()
         ctxt.register_action( "HELP_KEYBINDINGS" );
     }
 
+    std::set<long> character_blacklist = std::set<long>();
+    character_blacklist.insert( '+' );
+    character_blacklist.insert( '-' );
+    character_blacklist.insert( '=' );
+    character_blacklist.insert( KEY_ESCAPE );
+
     std::string hotkeys = ctxt.get_available_single_char_hotkeys( display_help_hotkeys );
+    std::vector<std::string> filtered_registered_actions = org_registered_actions;
+    std::string filter_phrase = "";
+    std::string action = "";
```

Comment:
You don't need to initialize string variables with an empty string. A default constructed string is automatically empty. The ` = ""` is redundant. The same goes for the set.